tools/bump_version.sh

```diff
@@ -94,6 +99,129 @@ validate_bump_type() {
+update_all_docker_version_refs() {
+    new_version=$(cat ${DOCKER_IMAGE_VERSION_PATH})
+
+    # Update Docker image versions in README files
+    update_image_versions ${ASTER_SRC_DIR}/README.md
+    update_image_versions ${ASTER_SRC_DIR}/README_CN.md
+    update_image_versions ${ASTER_SRC_DIR}/README_JP.md
+    update_image_versions ${SCRIPT_DIR}/docker/README.md
+    update_image_versions ${DOCS_DIR}/src/kernel/intel_tdx.md
+
+    # Update Docker image versions in workflows
+    WORKFLOWS=$(find "${ASTER_SRC_DIR}/.github/workflows/" -type f -name "*.yml")
+    for workflow in $WORKFLOWS; do
+        update_image_versions $workflow
+    done
+
+    # Update Docker image versions in the documentation
+    GET_STARTED_PATH=${ASTER_SRC_DIR}/docs/src/kernel/README.md
+    update_image_versions $GET_STARTED_PATH
+}
+
+# Update project dependencies (Cargo.toml and Cargo.lock)
+update_project_dependencies() {
+    # Update the versions in Cargo.toml
+    update_package_version ${OSTD_TEST_CARGO_TOML_PATH}
+    update_package_version ${OSTD_MACROS_CARGO_TOML_PATH}
+    update_package_version ${OSTD_CARGO_TOML_PATH}
+    update_package_version ${LINUX_BOOT_PARAMS_CARGO_TOML_PATH}
+    update_package_version ${LINUX_BZIMAGE_BUILDER_CARGO_TOML_PATH}
+    update_package_version ${LINUX_BZIMAGE_SETUP_CARGO_TOML_PATH}
+    update_package_version ${OSDK_CARGO_TOML_PATH}
+    update_package_version ${OSDK_TEST_RUNNER_CARGO_TOML_PATH}
+    update_package_version ${OSDK_FRAME_ALLOCATOR_CARGO_TOML_PATH}
+    update_package_version ${OSDK_HEAP_ALLOCATOR_CARGO_TOML_PATH}
+
+    update_dep_version ${OSDK_TEST_RUNNER_CARGO_TOML_PATH} ostd
+    update_dep_version ${OSDK_FRAME_ALLOCATOR_CARGO_TOML_PATH} ostd
+    update_dep_version ${OSDK_HEAP_ALLOCATOR_CARGO_TOML_PATH} ostd
+    update_dep_version ${OSTD_CARGO_TOML_PATH} ostd-test
+    update_dep_version ${OSTD_CARGO_TOML_PATH} linux-boot-params
+    update_dep_version ${OSTD_CARGO_TOML_PATH} ostd-macros
+    update_dep_version ${LINUX_BZIMAGE_SETUP_CARGO_TOML_PATH} linux-boot-params
+    update_dep_version ${OSDK_CARGO_TOML_PATH} linux-bzimage-builder
+
+    # Automatically bump Cargo.lock files
+    cargo update -p aster-nix --precise $new_version # For Cargo.lock
+    cd osdk && cargo update -p cargo-osdk --precise $new_version # For osdk/Cargo.lock
+}
+
+# Synchronize project version to Docker version (update VERSION)
+sync_project_version() {
+    new_version=$(cat ${DOCKER_IMAGE_VERSION_PATH} | cut -d'-' -f1)
+    current_version=$(cat ${VERSION_PATH})
+    if [ -z "$new_version" ] || [ -z "$current_version" ]; then
+        echo "Error: Version string is empty."
+        exit 1
+    fi
+
+    # Check if versions are equal
+    if [ "$new_version" = "$current_version" ]; then
+        echo "Versions are equal. No action needed."
+        exit 0
+    fi
+
+    # Compare semantic versions
+    lower_version=$(printf '%s\n' "$new_version" "$current_version" | sort -V | head -n1)
```

Comment:
No, we can't compare semantic versions as if they are plain text. For example, "0.9.0" < "0.10.0" in semantic versioning, but "0.10.0" < "0.9.0" when compared as strings. I guess that implementing comparison between semantic versions would be painful. If so, add a TODO here. And later, we can install in the next Docker image the right CLI tool to do this.

Reply:
I was wrong. `sort -V` is smart to handle semantic version numbers.